### PR TITLE
update to 1.0.12

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**
update to 1.0.12

- [requirements](https://github.com/snowflakedb/snowflake-ml-python/blob/1.0.12/requirements.yml)

### CODE REVIEW

**Comment by M-Waszkiewicz-Anaconda** [comment]
All python version are passing on windows custom host, see 
```
(base) %SRC_DIR%>conda activate "%PREFIX%"
import: 'snowflake.ml'
import: 'snowflake.ml.fileset'
import: 'snowflake.ml.model'
import: 'snowflake.ml.modeling'
import: 'snowflake.ml.registry'
import: 'snowflake.ml.utils'
import: 'snowflake.ml'
import: 'snowflake.ml.fileset'
import: 'snowflake.ml.model'
import: 'snowflake.ml.modeling'
import: 'snowflake.ml.registry'
import: 'snowflake.ml.utils'

Resource usage statistics from testing snowflake-ml-python:
   Process count: 1
   CPU time: Sys=0:00:00.0, User=-
   Memory: 4.3M
   Disk usage: 1.5K
   Time elapsed: 0:00:02.1


TEST END: C:\Users\48667\miniconda3\conda-bld\win-64\snowflake-ml-python-1.0.12-py311h2bbff1b_0.tar.bz2
Renaming work directory 'C:\Users\48667\miniconda3\conda-bld\snowflake-ml-python_1700135414178\work' to 'C:\Users\48667\miniconda3\conda-bld\snowflake-ml-python_1700135414178\work_moved_snowflake-ml-python-1.0.12-py311h2bbff1b_0_win-64_main_build_loop'
INFO:conda_build.utils:Renaming work directory 'C:\Users\48667\miniconda3\conda-bld\snowflake-ml-python_1700135414178\work' to 'C:\Users\48667\miniconda3\conda-bld\snowflake-ml-python_1700135414178\work_moved_snowflake-ml-python-1.0.12-py311h2bbff1b_0_win-64_main_build_loop'
shutil.move(work)=C:\Users\48667\miniconda3\conda-bld\snowflake-ml-python_1700135414178\work, dest=C:\Users\48667\miniconda3\conda-bld\snowflake-ml-python_1700135414178\work_moved_snowflake-ml-python-1.0.12-py311h2bbff1b_0_win-64_main_build_loop)
INFO:conda_build.utils:shutil.move(work)=C:\Users\48667\miniconda3\conda-bld\snowflake-ml-python_1700135414178\work, dest=C:\Users\48667\miniconda3\conda-bld\snowflake-ml-python_1700135414178\work_moved_snowflake-ml-python-1.0.12-py311h2bbff1b_0_win-64_main_build_loop)
WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(188): Could not remove or rename C:\Users\48667\miniconda3\conda-bld\snowflake-ml-python_1700135414178\_build_env\Library\lib\modules.  Please remove this file manually (you may need to reboot to free file handles)
# Automatic uploading is disabled
# If you want to upload package(s) to anaconda.org later, type:


# To have conda build upload to anaconda.org automatically, use
# $ conda config --set anaconda_upload yes
anaconda upload ^
    C:\Users\48667\miniconda3\conda-bld\win-64\snowflake-ml-python-1.0.12-py311h2bbff1b_0.tar.bz2
anaconda_upload is not set.  Not uploading wheels: []

INFO :: The inputs making up the hashes for the built packages are as follows:
{
  "snowflake-ml-python-1.0.12-py311h2bbff1b_0": {
    "recipe": {
      "c_compiler": "vs2017",
      "channel_targets": "defaults",
      "target_platform": "win-64"
    }
  }
}


####################################################################################
Resource usage summary:

Total time: 0:17:15.7
CPU usage: sys=0:00:53.7, user=0:02:47.9
Maximum memory usage observed: 967.9M
Total disk usage observed (not including envs): 15.5G


####################################################################################
Source and build intermediates have been left in C:\Users\48667\miniconda3\conda-bld.
There are currently 21 accumulated.
```

File: recipe/meta.yaml [hunk 1]
```diff
@@ -34,6 +34,7 @@ requirements:
     - anyio >=3.5.0,<4
     - cloudpickle >=2.0.0
     - fsspec >=2022.11,<2024
+    - importlib_resources >=5.1.4,<6
     - numpy >=1.23,<2
     - packaging >=20.9,<24
     - pandas >=1.0.0,<2
```
Comment (boldorider4) [inline recipe/meta.yaml:40]: it would be cool to fix
`    - pytimeparse>=1.1.8,<2`
so that linter error goes away

Comment (boldorider4) [reply recipe/meta.yaml:40]: actually you just need to remove this line.

**Review by boldorider4 — COMMENTED** [review]
It would be cool to add upstream tests, but I understand that this was released before without them.

**Comment by M-Waszkiewicz-Anaconda** [comment]
> It would be cool to add upstream tests, but I understand that this was released before without them.

we tried it with previous releases, most of them failing as they are integration tests supposed to be run in snowflake env

File: recipe/meta.yaml (hunk 1 above)
Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:40]: we need it 
https://github.com/snowflakedb/snowflake-ml-python/blob/b938743b2b7534c97b56ba5be9b926184533ad7d/requirements.yml#L268

Comment (M-Waszkiewicz-Anaconda) [reply recipe/meta.yaml:40]: ahh ok, it is dobuled

**Review by boldorider4 — APPROVED** [review]

**Review by cbouss — COMMENTED** [review]
`- transformers >=4.29.2,<5` --> `- transformers >=4.32.1,<5`
https://github.com/snowflakedb/snowflake-ml-python/compare/1.0.11...1.0.12#diff-5edc2578758a214f70cba8305755573302cfc4a2125cc756e62949bb3eb9f924R240

**Comment by M-Waszkiewicz-Anaconda** [comment]
> `- transformers >=4.29.2,<5` --> `- transformers >=4.32.1,<5` [snowflakedb/snowflake-ml-python@1.0.11...1.0.12#diff-5edc2578758a214f70cba8305755573302cfc4a2125cc756e62949bb3eb9f924R240](https://github.com/snowflakedb/snowflake-ml-python/compare/1.0.11...1.0.12#diff-5edc2578758a214f70cba8305755573302cfc4a2125cc756e62949bb3eb9f924R240)

Good catch ! , Done

File: recipe/meta.yaml [hunk 2]
```diff
@@ -60,7 +61,7 @@ requirements:
     - tensorflow >=2.9,<3,!=2.12.0
     - tokenizers>=0.10,<1
```
Comment (akabanovs) [inline recipe/meta.yaml:62]: Add a space between the name and the version range.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -58,9 +60,9 @@ requirements:
     - mlflow >=2.1.0,<2.4
     - sentencepiece >=0.1.95,<0.2
     - tensorflow >=2.9,<3,!=2.12.0
-    - tokenizers>=0.10,<1
+    - tokenizers >=0.10,<1
     - torchdata >=0.4,<1
-    - transformers >=4.29.2,<5
+    - transformers >=4.32.1,<5
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:65]:
```suggestion
    - transformers >=4.32.1,<5
    - vllm >=0.2.1.post1,<1
```
https://github.com/snowflakedb/snowflake-ml-python/blob/1.0.12/snowflake/ml/requirements.bzl#L15C10-L15C30

**Review by skupr-anaconda — APPROVED** [review]
LGTM but `run_constrained`

**Pull request merged**